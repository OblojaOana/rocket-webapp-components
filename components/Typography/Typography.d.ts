import { TypographyProps as BaseProps } from '@material-ui/core/Typography'

export interface TypographyProps extends BaseProps {

    /**
     * If provided, a text will appear on hover
     */
    tooltip?: string;

    /**
     * Applies the theme typography styles.
     * @default 'inherit'
     */
    variant?: 'inherit' |
    'h1' |
    'h2' |
    'h3' |
    'h4' |
    'h5' |
    'h6' |
    'subtitle1' |
    'subtitle2' |
    'body1' |
    'body2' |
    'caption' |
    'button' |
    'overline'

    /**
     * The color of the component
     */
    color?: 'initial' |
    'inherit' |
    'primary' |
    'secondary' |
    'textPrimary' |
    'textSecondary' |
    'error'

    /**
     * Set the text-align on the component.
     */
    align?: 'inherit' |
    'left' |
    'center' |
    'right' |
    'justify'

    /**
     * Demos:
     * 
     * - https://bit.dev/totalsoft_oss/react-mui/typography
     * 
     */

}

export default function Typography(props: TypographyProps): JSX.Element;