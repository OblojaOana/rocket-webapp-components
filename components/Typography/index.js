import Typography from '../../core/Typography';
export default Typography;