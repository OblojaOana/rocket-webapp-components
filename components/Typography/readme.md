##### Copy this snapshot in the overview panel

```js
import React from "react";
import Typography from "@bit/totalsoft_oss.react-mui.typography";

export default <Typography tooltip="Tooltip">Text to be displayed</Typography>;
```
