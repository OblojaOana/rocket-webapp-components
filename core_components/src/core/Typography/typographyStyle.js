const typographyStyle = theme => ({
    defaultFont: {
        fontFamily: theme.typography.fontFamily,
        textTransform: 'none'
    }
})

export default typographyStyle;