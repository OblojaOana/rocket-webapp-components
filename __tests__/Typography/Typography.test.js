import React from "react";
import renderer from "react-test-renderer";
import Typography from "../../components/Typography/Typography";
import { mount } from 'enzyme';

describe("Typography", () => {
  it("Snapshot", () => {
    const component = renderer.create(
      <Typography tooltip="Tooltip" variant="body1" align="center">
        Text to be displayed
      </Typography>
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it("varaint property shoulb be equalt to the default value provided", () => {
    const wrapper = mount(
      <Typography tooltip="Tooltip" align="center">
        Text to be displayed
      </Typography>
    );

    expect(wrapper.props().variant).toBe("inherit");
  });
});
