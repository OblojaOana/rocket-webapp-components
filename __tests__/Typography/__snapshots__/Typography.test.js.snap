// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Typography render correctly Typography component 1`] = `
<p
  aria-describedby={null}
  className="MuiTypography-root makeStyles-defaultFont-1 MuiTypography-body1 MuiTypography-alignCenter"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Tooltip"
>
  Text to be displayed
</p>
`;
